luma/lcd/device.py

```diff
@@ -757,3 +760,234 @@ def contrast(self, value):
+        Command DISPLAYOFF                                                  40μs
+        Command CLEAR                                                       40μs
+        Command ENTRY ; No shift, incrementing display                      40μs
+
+        4bit mode
+        Action                                                            Delay
+        ----------------------------------------------------------------  -------
+        PowerOn                                                           40.0ms
+        Command 0x03 ; 4 bit write                                         4.1ms
+        Command 0x03 ; 4 bit write                                         100μs
+        Command 0x32                                                        40μs
+        Command FUNCTIONSET|DL4|(LINES2 or LINES1)|(CHAR5x8 or CHAR5x10)    40μs
+        Command DISPLAYOFF                                                  40μs
+        Command CLEAR                                                       40μs
+        Command ENTRY ; No shift, incrementing display                      40μs
+
+        Device is now ready to receive data
+        """
+        sleep(1e-3 * 100)
+        if self._bitmode == 8:
+            self.command(0x30, exec_time=1e-3 * 10)
+            self.command(0x30, exec_time=1e-6 * 200)
+            self.command(0x30)
+        else:
+            ''' In case of stubborn display place in 8 bit mode three times and
+                and then try to get it into four bit mode '''
+            for _ in range(3):
+                self.command(0x03, exec_time=1e-3 * 10, only_low_bits=True)
+                self.command(0x03, exec_time=1e-3 * 10, only_low_bits=True)
+                self.command(0x33, exec_time=1e-3 * 100)
+            self.command(0x03, exec_time=1e-3 * 10, only_low_bits=True)
+            self.command(0x03, exec_time=1e-6 * 200, only_low_bits=True)
+            self.command(0x32)
+
+        dl = self._const.DL8 if self._bitmode == 8 else self._const.DL4
+        self.command(self._const.FUNCTIONSET | dl | self._const.LINES2)
+        self.command(self._const.DISPLAYOFF)  # Set Display Off
+        self.command(self._const.ENTRY)  # Set entry mode to direction right, no shift
+        self.command(self._const.DISPLAYON, exec_time=1e-3 * 100)  # Turn Display back on
+
+    def display(self, image):
+        """
+        Takes a 1-bit :py:mod:`PIL.Image` and converts it to text data
+        by reversing from glyphs from the image back to the correct
+        value from the displays font table.
+
+        If needed, it will create custom characters if a glyph is not found
+        within the font table
+
+        .. note:
+            Most displays have limited memory to support custom characters
+            and typically can only support 8 at any one time.  If this is
+            exceeded, the remaining unmatched characters will be replaced by
+            the undefined character.
```

Comment:
undefined should probaby be in double backticks? ```undefined```

Reply:
Changed as suggested